src/main/java/lotto/model/ScoreMap.java

```diff
@@ -0,0 +1,62 @@
+package lotto.model;
+
+import lotto.model.config.LottoConfig;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class ScoreMap {
+    private Map<Integer, Integer> scoreMap;
+
+    public ScoreMap() {
+        this.scoreMap = new HashMap<Integer, Integer>(){{
+            put(0, 0);
+            put(1, 0);
+            put(2, 0);
+            put(3, 0);
+            put(4, 0);
+            put(5, 0);
+            put(6, 0);
+        }};
+    }
+
+    public Set<Map.Entry<Integer, Integer>> getEntrySet(){
+        return scoreMap.entrySet();
+    }
+
+    public void updateWithNumMatched(int numMatched) {
+        scoreMap.put(numMatched, scoreMap.get(numMatched) + 1);
+    }
+
+    public int calculateReward() {
+        int totalProfit = 0;
+        for( Map.Entry<Integer, Integer> lottoResult : scoreMap.entrySet() ){
+            int numMatched = lottoResult.getKey();
+            int numMatchedLottoNumbers = lottoResult.getValue();
+            totalProfit += calculateEachReward(numMatched, numMatchedLottoNumbers);
+        }
+        return totalProfit;
+    }
+
+    private int calculateEachReward(int key, int value){
+        if(LottoConfig.winningRewards.get(key) == 0 || value == 0){
```

Comment:
```suggestion
        if (LottoConfig.winningRewards.get(key) == 0 || value == 0) {
```
매직넘버가 추출되면 좀 더 코드 읽기가 수월할 것 같습니다 🙇 
> https://hoonmaro.tistory.com/44

Reply:
매직넘버를 제거하였습니다.